src/client/actions.ts

```diff
@@ -0,0 +1,132 @@
+ *         // the application will not receive a "notification-action" event when the button is clicked
+ *         {
+ *             title: 'Dismiss',
+ *             iconUrl: 'https://www.example.com/cancel.png'
+ *         },
+ *
+ *         // A button that will schedule another reminder for 5 minutes from now. Since the application will be
+ *         // responsible for snoozing the event, it will need to know about the user clicking this button. By setting
+ *         // 'onClick', the service will raise a "notification-action" event when this button is clicked, and will
+ *         // pass the value of 'onClick' as the 'result' field within the event.
+ *         {
+ *             title: 'Snooze for 5 minutes',
+ *             iconUrl: 'https://www.example.com/timer.png',
+ *             onClick: {
+ *                 action: 'schedule-reminder',
+ *                 intervalMs: 5 * 60 * 1000
+ *             }
+ *         }
+ *     ]
+ * });
+ *
+ * // Create a listener that will be called for each action
+ * // Note: This handler will be used for ALL actions, from ALL notifications that are created by this application.
+ * addEventListener('notification-action', (event: NotificationActionEvent) => {
+ *     const {result, notification} = event;
+ *
+ *     if (result.action === 'view-event') {
+ *         // Open a window with full details of the associated event
+ *         openEventDetails(notification.customData.eventId, result.target);
+ *     } else if (result.action === 'schedule-reminder') {
+ *         // Schedule a new notification
+ *         scheduleReminder(notification.customData.eventId, Date.now() + result.intervalMs);
+ *     } // Etc...
+ * });
+ * ```
+ *
+ * The example above uses `customData` to store details about the notification subject (in this case, a calendar
+ * event), and `onClick` actions to inform the application about how it should respond when the user interacts with the
+ * notification. This is our intended usage and recommended best-practice, but the service doesn't require applications
+ * to follow this convention - application developers are free to decide how to manage notification state.
+ *
+ * Within the `notification-action` handler, the application must be able to understand which notification is being
+ * handled, and how to decide what it should do next. The example above uses an application-defined `action` field to
+ * determine the correct action, but the notification's `id`, `category`, `customData` and other fields are also useful
+ * selectors.
+ *
+ * @module Actions
+ */
+
+/**
+ * Denotes a field as being an action. Defining this field (with a non-`undefined` value) will result in actions being
+ * generated and sent back to the source application when the corresponding event happens.
+ *
+ * For example, providing a value for the `onClick` field of {@link ButtonOptions} will result in a
+ * `notification-action` event being raised when that button is clicked.
```

Comment:
Go with 'fired' rather than raised - so it's actions are raised, events are fired